### ISSUE TITLE
vscode requests breakpointLocations on every step

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#86850]
@weswigham pointed this out.

Every time I step while debugging in vscode, the debug adapter gets a breakpointLocations request. We should only make this request when we need to, because it could be expensive.  While debugging tests in the TS codebase, there is a case where sometimes the DA's request to the runtime takes a few seconds to complete when it is triggered at the same time as a step. The result is that each step takes about 3s, which is not a good experience. The file we are debugging is 11MB which probably accounts for part of the slowness.

**Comment by roblourens** [comment]
@weswigham here is a version of the DA that has breakpointLocations disabled, if you want to verify that this fixes it

[node-debug2-1.41.3.vsix.zip](https://github.com/microsoft/vscode/files/3958042/node-debug2-1.41.3.vsix.zip)

**Comment by weswigham** [comment]
Errr, it actually doesn't:

Check out line 1529:
[debugadapter.txt](https://github.com/microsoft/vscode/files/3958065/debugadapter.txt)

there's still a 3s delay between `resume` and the `pause` on the next line (and the line in question in this log is just an `array.push`, which shouldn't be a 3s operation).

**Comment by weswigham** [comment]
Yeah - definitely a v8 thing - if I revert to `node` 8 (vs 13), perf is fine. Lemme see if I can approximately track when it regressed.

**Comment by weswigham** [comment]
Done. It's a regression in `node 13.2.0`. I haven't checked what v8 versions/changes that corresponds to, yet.

**Comment by roblourens** [comment]
I filed https://github.com/nodejs/node/issues/30926, also FYI @connor4312

**Comment by connor4312** [comment]
May be related to https://github.com/microsoft/vscode-js-debug/issues/138 as well (I tested on 12 and marked as CNR, could have been caused by testing on >13.2)

**Comment by roblourens** [comment]
I don't think so. There aren't many variables being sent around.

**Comment by weswigham** [comment]
Looks like this will be fixed by a new version of v8 (version 8.0 or above) in node 14+. I've checked with a canary build, specifically https://nodejs.org/download/v8-canary/v14.0.0-v8-canary2020011995087de57b/

**Comment by connor4312** [comment]
Great to hear!
